### DISCUSSION TITLE
What can I do if I'm stuck in a single database and I'm not allowed to even create SCHEMA objects

### DISCUSSION THREAD

**Discussion by ghost** [littleK0i/SnowDDL#69] [Q&A] [answered]
Can I push everything down inside one or more pre-existing schemas of a single database? The snowflake I am in is VERY locked down.

**Comment by littleK0i** [comment, marked as answer]
At this moment SnowDDL normally requires the ability to create schemas even in [SingleDB mode](https://docs.snowddl.com/single-db/overview).

But you may try to create schema(s) manually and explicitly grant it to SnowDDL admin role.

```
CREATE SCHEMA ... ;
GRANT OWNERSHIP ON SCHEMA ... TO SNOWDDL_ADMIN;
```

SnowDDL should check existence and parameters of schema object. But that's it. It should not touch existing schema if it was configured correctly.

You'll have to create future grants for this schema manually as well.

**Reply by ghost** [reply]
Thank you so much!!
